src/utils.c

```diff
@@ -849,8 +849,14 @@ ts_get_integer_now_func(const Dimension *open_dim)
 
 	if (strlen(NameStr(open_dim->fd.integer_now_func)) == 0 &&
 		strlen(NameStr(open_dim->fd.integer_now_func_schema)) == 0)
-		ereport(ERROR,
-				(errcode(ERRCODE_UNDEFINED_FUNCTION), (errmsg("integer_now function not set"))));
+	{
+		if (fail_if_not_found)
+			ereport(ERROR,
+					(errcode(ERRCODE_UNDEFINED_FUNCTION),
+					 (errmsg("integer_now function not set"))));
+		else
+			return now_func;
+	}
```

Comment:
This doesn't get hit because again the callers do additional checks elsewhere early on for a valid `now_func`